language/stdlib/modules/LibraWriteSetManager.move

```diff
@@ -63,6 +64,7 @@ module LibraWriteSetManager {
         // The below code uses direct abort codes as per contract with VM.
         let sender = Signer::address_of(account);
         assert(sender == CoreAddresses::LIBRA_ROOT_ADDRESS(), PROLOGUE_EINVALID_WRITESET_SENDER);
+        assert(Roles::has_libra_root_role(account), PROLOGUE_EINVALID_WRITESET_SENDER);
```

Comment:
It seems in some places we assume the `libra_root_role` has a fixed address requirement, so we may want to do the same here? (i.e., remove the `assert(sender == CoreAddresses::LIBRA_ROOT_ADDRESS())`). Not crucial for this PR but marking for myself and others when going through to separate out operational constraints from role constraints.

Reply:
Just call Roles::assert_libra_root here which contains both address and role check. Then this reason here can be removed.

Reply:
This actually needs to be an in-place assert (`assert(Roles::has_libra_root_role(..), PROLOGUE_EINVALID_WRITESET_SENDER)`) since the VM expects a specific error code from the prologue here.

Reply:
`Roles::assert_libra_root` checks the address, but `assert(Roles::has_libra_root_role(..), ...)` does not. @tzakian, do we still want to remove the address check (i.e., `assert(sender == CoreAddresses::LIBRA_ROOT_ADDRESS())`)?

Reply:
I keep the code as it is. Instead of `assert_libra_root`, it needs to use `has_libra_root_role` for the custom error code for VM, and need to check the LR address because `has_libra_root_role` does not.